docs/topics/settings.rst

```diff
@@ -234,6 +250,61 @@ example, proper setting names for a fictional robots.txt extension would be
 ``ROBOTSTXT_ENABLED``, ``ROBOTSTXT_OBEY``, ``ROBOTSTXT_CACHEDIR``, etc.
 
 
+Special settings
+================
+
+The following settings work slightly differently than all other settings.
+
+.. _pre-crawler-settings:
+
+Pre-crawler settings
+--------------------
+
+**Pre-crawler settings** are settings used before the
+:class:`~scrapy.crawler.Crawler` object is created.
+
+These settings cannot be :ref:`set from a spider <spider-settings>`.
+
+These settings are :setting:`SPIDER_LOADER_CLASS` and settings used by the
+corresponding :ref:`component <topics-components>`, e.g.
+:setting:`SPIDER_MODULES` and :setting:`SPIDER_LOADER_WARN_ONLY` for the
+default component.
+
+
+.. _reactor-settings:
+
+Reactor settings
+----------------
+
+**Reactor settings** are settings tied to the :doc:`Twisted reactor
+<twisted:core/howto/reactor-basics>`.
+
+These settings can be defined from a spider. However, because only 1 reactor
+can be used per process, these settings cannot use a different value per spider
+when :ref:`running multiple spiders in the same process
+<run-multiple-spiders>`.
+
+In general, if different spiders define different values, the first defined
+value is used. However, if two spiders request a different reactor, an
```

Comment:
Hm, I didn't know that :( I was expecting the last value to be used, e.g. for dns options, or for the threadpool size.

Reply:
OK, so two fun facts:

* `DNS_RESOLVER` and `REACTOR_THREADPOOL_MAXSIZE` are only honored by `CrawlerProcess`, if you use `CrawlerRunner` they will be ignored.
* `DNS_RESOLVER` and `REACTOR_THREADPOOL_MAXSIZE` are read directly from the settings passed to `CrawlerProcess`, which are *not* modified by `self.spidercls.update_settings()` in `Crawler.__init__()` so they cannot be changed via per-spider settings.

As for "the first value is used", it's true for `TWISTED_REACTOR` and `ASYNCIO_EVENT_LOOP` because the first `CrawlerProcess.crawl()` calls `install_reactor()`.

Should we make an issue to clarify the docs about this further? Should we keep the code as is?

Reply:
> Should we make an issue to clarify the docs about this further? 

+1, *maybe* a good first issue.

> Should we keep the code as is?

I don’t think it is urgent to change it, but it would be nice to create a separate, long-term ticket to discuss improvements to this, in case someone has a strong proposal.